auth/auth.go

```diff
@@ -129,21 +149,35 @@ func (p Auth) getRoleARN() (arn *string, e error) {
 //
 // The returned session is capable of automatically refreshing creds as needed
 // by using a private TokenFetcher helper.
-//
 func (p Auth) GetAWSSession() (awsSession *session.Session, e error) {
-
-	roleArn, err := p.getRoleARN()
-	if err != nil {
-		return nil, err
+	var config *aws.Config
+
+	fetcher := &authTokenFetcher{p.nameSpace, p.svcAcc, p.podName, p.k8sClient, p.usePodIdentity}
+
+	if p.usePodIdentity {
```

Comment:
Instead of doing this, you could write an interface for providing credentials and have two implementations of that interface, one for PIA and one for IRSA.